resources/qml/Sidebar.qml

```diff
@@ -256,6 +256,44 @@ Rectangle
         }
     }
 
+    Button
+    {
+        id: toggleFilterButton
+
+        anchors.right: parent.right
+        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+        anchors.top: headerSeparator.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+
+        height: settingsModeSelection.height
+        width: visible ? height : 0
```

Comment:
A visible item still has a width and a height. Anything that layouts based on an item's width or height will thus still use the original width and height of the item. Strictly speaking, I do not think that is the case here but changing width here will make the expression above simpler.